CONTRIBUTING.md

```diff
@@ -0,0 +1,77 @@
+# Contributing to Bicep
+We are very happy to accept community contributions to Bicep, whether those are [Pull Requests](#pull-requests), [Example Files](#example-files), [Feature Suggestions](#feature-suggestions) or [Bug Reports](#bug-reports)! Please note that by participating in this project, you agree to abide by the [Code of Conduct](./CODE_OF_CONDUCT.md), as well as the terms of the [CLA](#cla).
+
+## Getting Started
+* If you haven't already, you will need [dotnet](https://dotnet.microsoft.com/download) and [node + npm](https://nodejs.org/en/download/) installed locally to build and run this project.
+* You are free to work on Bicep on any platform using any editor, but you may find it quickest to get started using [VSCode](https://code.visualstudio.com/Download) with the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp).
+* Fork this repo (see [this forking guide](https://guides.github.com/activities/forking/) for more information).
+* Checkout the repo locally with `git clone git@github.com:{your_username}/Bicep.git`.
+* Build the .NET solution with `dotnet build`.
+
+## Developing
+### Components
+The Bicep solution is comprised of the following main components:
+* **Bicep CLI** (`src/Bicep.Cli`): the `bicep` CLI exectuable.
+* **Bicep Language Server** (`src/Bicep.LangServer`): the LanguageServer used by the VSCode extension for parsing and providing information about a Bicep file.
+* **Bicep Core** (`src/Bicep.Core`): the library containing the majority of the Biceo compiler code.
+* **Bicep VSCode Extension** (`src/vscode-bicep`): the VSCode extension itself. This is mostly a thin wrapper around the Bicep Language Server.
+* **Playground** (`src/playground`): the web-based playground hosted at `https://aka.ms/bicepdemo`.
+* A number of different test suites.
+
+### Running the tests
+* You can use the following command to run the full Bicep test suite:
+    * `dotnet test`
+
+### Running the Bicep VSCode extension
+* On the first run, you'll need to ensure you have installed all the npm packages required by the Bicep VSCode extension with the following:
+    * `cd src/bicep-vscode`
+    * `npm i`
+* In the [VSCode Run View](https://code.visualstudio.com/Docs/editor/debugging), select the "Bicep VSCode Extension" task, and press the "Start" button. This will launch a new VSCode window with the Bicep extension and LanguageServer containing your changes.
+* If you want the ability to put breakpoints and step through the C# code, you can also use the "Attach" run configuration once the extension host has launched, and select the Bicep LanguageServer process.
```

Comment:
Unfortunately the process is dotnet.exe on windows ☹. I wonder if we should log the PID somewhere in the VS code extension.

Reply:
![image](https://user-images.githubusercontent.com/38542602/94491667-7d620080-019d-11eb-82eb-ab50a0359da4.png)
This works in Windows

Reply:
Not in Visual Studio though.

Reply:
Ah... I thought VS gave you a similar view with the cmdline args displayed. Logging the PID would for sure be useful.